certora/applyMunging.patch

```diff
@@ -123,3 +136,34 @@ diff -ruN MetaMorpho.sol MetaMorpho.sol
          DECIMALS_OFFSET = uint8(uint256(18).zeroFloorSub(IERC20Metadata(_asset).decimals()));
  
          _checkTimelockBounds(initialTimelock);
+@@ -338,6 +340,9 @@
+             seen[prevIndex] = true;
+ 
+             newWithdrawQueue[i] = id;
++
++            // HARNESS
++            withdrawRank[id] = i + 1;
```

Comment:
If we use `withdrawIndex` instead of `withdrawRank`, then we can't state the withdraw index correctness property for the index 0. This is because all values are 0 by default in a mapping, so you can't say if it's 0 because `id` is indeed at index 0, or if it is 0 by default. Notably, in the `enabledIsInWithdrawQueue` rule, your witness would be `withdrawIndex(id)` directly, but the prover would give you a counterexample if it returns 0

Reply:
get it